docs/source/design.rst

```diff
@@ -12,6 +12,18 @@ or if you are just looking to access some information that is not currently expo
 .. contents:: Contents
    :local:
 
+.. _initialization:
+
+Initialization
+**************
+
+When the IP Address of a device is known, it can be connected with :meth:`~kasa.SmartDevice.connect()`.
+If the IP Address is not known, it can be discovered using :func:`~kasa.Discover.discover_devices`.
```

Comment:
```suggestion
If the IP Address is not known, it can be discovered using :func:`~kasa.Discover.discover`.
```
Note the wrong function name. I would also swap this and the previous sentence, so something like:
> Use <discover> to perform broadcast, udp-based discovery on the network. This will return you a list of device instances based on the discovery replies.
If the device's host is already known, you can use <connect> to construct a device instance.

Reply:
I think it looks good now, but I'm really jet lagged so I'll check it again in the morning. Than I think we should get a release out to address the linked issues in HA

Reply:
Ah, it looked fine to me so I went and merged it already... I can do release prep on the weekend if that works.

Reply:
@bdraco I think it might be prudent to keep HA doing `discover_single` instead of direct tcp connection for the next HA release. I understand there are some issues at the moment with subnets and udp but the klap changes might not fully work with TCP connect and we may need to sync up on the error handling in HA.

Reply:
I'd rather get the users who are using the integration today fixed before we focus on adding new things.

Reply:
Its ok if we have to refactor the integration for KLAP in another release. That can happen rather quickly.

Reply:
I don't think this changes anything regarding to klap, or auth support in general. The discovery will still done using the discovery protocol and what changes is only that homeassistant can optimize after-discovery accesses when it knows the connection parameters.

So basically the discovered (or inputted) connection settings are stored inside the config entry, which are then used for future connections. So for authenticated devices, the integration will need to be modified to catch the exception on invalid creds / creds needed and that should be enough. Or am I mistaken?

Reply:
That seems like a great path forward to me unless I'm missing something as well 👍

Reply:
Agreed. Sorry, I just meant that we should incorporate that error handling and connection parameters before going ahead with the draft HA PR as-is. 

It’s not essential because as @bdraco says, it’s new functionality, but atm it could raise errors for anyone who adds a klap device via discovery then restarts. 

If we can fix rapidly in HA then not so much of a problem.

Reply:
I added a commit to https://github.com/home-assistant/core/pull/104213/commits/4d3b86b77b46230ad0d069efc89047c153d8e139 to make sure klap devices don't start a discovery flow so they don't appear before we have code in place to support them in HA